tiledb/sm/misc/utils.cc

```diff
@@ -451,14 +458,48 @@ bool overlap(const T* a, const T* b, unsigned dim_num) {
   return true;
 }
 
+template <class T>
+bool overlap(const T* a, const T* b, unsigned dim_num, bool* a_contains_b) {
+  for (unsigned i = 0; i < dim_num; ++i) {
+    if (a[2 * i] > b[2 * i + 1] || a[2 * i + 1] < b[2 * i])
+      return false;
+  }
+
+  *a_contains_b = true;
+  for (unsigned i = 0; i < dim_num; ++i) {
+    if (a[2 * i] > b[2 * i] || a[2 * i + 1] < b[2 * i + 1]) {
+      *a_contains_b = false;
+      break;
+    }
+  }
+
+  return true;
+}
+
+template <class T>
+void overlap(const T* a, const T* b, unsigned dim_num, T* o, bool* overlap) {
+  // Get overlap range
+  *overlap = true;
+  for (unsigned int i = 0; i < dim_num; ++i) {
+    o[2 * i] = MAX(a[2 * i], b[2 * i]);
+    o[2 * i + 1] = MIN(a[2 * i + 1], b[2 * i + 1]);
+    if (o[2 * i] > b[2 * i + 1] || o[2 * i + 1] < b[2 * i]) {
+      *overlap = false;
+      break;
+    }
+  }
+}
+
 template <class T>
 double coverage(const T* a, const T* b, unsigned dim_num) {
   double c = 1.0;
-  for (unsigned i = 0; i < dim_num; ++i)
-    if (std::is_integral<T>::value)
-      c *= double(a[2 * i + 1] - a[2 * i] + 1) / (b[2 * i + 1] - b[2 * i] + 1);
-    else
-      c *= double(a[2 * i + 1] - a[2 * i]) / (b[2 * i + 1] - b[2 * i]);
+  auto add = double(std::is_integral<T>::value);
```

Comment:
Isn't this calculation problematic value is an integer and not representable by a double (> 2^53)?

Reply:
I could make this an integer, but later it is used in numbers that have to be `double` anyway. Coverage is an estimate anyway, so I don't expect any problems because of loss of accuracy here.